modules/apps/object/object-dynamic-data-mapping-form-field-type/src/test/java/com/liferay/object/dynamic/data/mapping/form/field/type/internal/attachment/AttachmentDDMFormFieldTemplateContextContributorTest.java

```diff
@@ -0,0 +1,186 @@
+import com.liferay.dynamic.data.mapping.model.DDMFormField;
+import com.liferay.dynamic.data.mapping.render.DDMFormFieldRenderingContext;
+import com.liferay.dynamic.data.mapping.test.util.BaseDDMFormFieldTemplateContextContributorTestCase;
+import com.liferay.object.dynamic.data.mapping.form.field.type.constants.ObjectDDMFormFieldTypeConstants;
+import com.liferay.portal.json.JSONFactoryImpl;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.repository.model.FileEntry;
+import com.liferay.portal.kernel.test.ReflectionTestUtil;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.upload.configuration.UploadServletRequestConfigurationProvider;
+import com.liferay.portal.kernel.util.MapUtil;
+import com.liferay.portal.test.rule.LiferayUnitTestRule;
+
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.mockito.Mockito;
+
+import org.skyscreamer.jsonassert.JSONAssert;
+
+/**
+ * @author Pedro Leite
+ */
+public class AttachmentDDMFormFieldTemplateContextContributorTest
+	extends BaseDDMFormFieldTemplateContextContributorTestCase {
+
+	@ClassRule
+	@Rule
+	public static final LiferayUnitTestRule liferayUnitTestRule =
+		LiferayUnitTestRule.INSTANCE;
+
+	@Before
+	@Override
+	public void setUp() throws Exception {
+		super.setUp();
+
+		_setUpDLAppLocalService();
+		_setUpDLURLHelper();
+		_setUpJSONFactory();
+		_setUpLanguage();
+		_setUpUploadServletRequestConfigurationProvider();
+
+		_ddmFormField.setDDMForm(getDDMForm());
+	}
+
+	@Test
+	public void testGetFileEntryProperties() throws Exception {
+		String contentUrl = RandomTestUtil.randomString();
+
+		_ddmFormField.setProperty("contentURL", contentUrl);
+
+		_ddmFormField.setProperty("localizedObjectField", true);
+
+		DDMFormFieldRenderingContext ddmFormFieldRenderingContext =
+			createDDMFormFieldRenderingContext();
+
+		Long value1 = RandomTestUtil.randomLong();
+		Long value2 = RandomTestUtil.randomLong();
+
+		ddmFormFieldRenderingContext.setValue(
+			JSONUtil.put(
+				"en_US", value1
+			).put(
+				"pt_BR", value2
+			).toString());
+
+		_mockFileEntry(_fileEntry1, value1);
+		_mockFileEntry(_fileEntry2, value2);
+
+		Map<String, Object> parameters =
+			_attachmentDDMFormFieldTemplateContextContributor.getParameters(
+				_ddmFormField, ddmFormFieldRenderingContext);
+
+		JSONAssert.assertEquals(
+			JSONUtil.put(
+				"en_US",
+				JSONUtil.put(
+					"contentURL", contentUrl
+				).put(
+					"title", String.valueOf(value1)
+				)
+			).put(
+				"pt_BR",
+				JSONUtil.put(
+					"contentURL", contentUrl
+				).put(
+					"title", String.valueOf(value2)
+				)
+			).toString(),
+			String.valueOf(parameters.get("fileEntryProperties")), false);
+	}
+
+	@Test
+	public void testGetLocalizedObjectFieldTrue() {
+		_ddmFormField.setDDMForm(getDDMForm());
```

Comment:
Do we need this line since we defined this on the `setUp()` method?

Reply:
Done!